### PR TITLE
Allow passing videoId == null to keep a player around without playing video

### PR DESCRIPTION
**Pull request by goto-bus-stop**
Note: this conflicts with #63, so if both PRs are accepted I'll rebase whichever
is merged last.

This patch accepts `videoId={null}` or to omit the `videoId` prop entirely. If
the `videoId` prop is empty, no video will be played, or the playing video will
be stopped.

This allows `react-youtube` to also represent the YouTube API's "stopped" state.
It's useful for cases where you don't know what you're going to play yet, but
where you do want to have the YouTube iframe around. Perhaps to reserve pixels
on the page, or to be able to load videos more quickly once you _do_ know what
to play, or to keep the `iframe` on the page to work around Chrome's [gesture
requirement for media playback](http://techdows.com/2015/08/future-chrome-versions-wont-autoplay-media-in-the-background-tabs.html).

### CODE REVIEW

**Comment by tjallingt** [comment]
I had noted that `videoId` was set to `isRequired` before this but ignored it for the time being...
I agree that it doesn't make much sense as you could also set `videoId` in the `opts`. Furthermore the YouTube Iframe API doesn't require you to specify a `videoId` in the first place (it will accept `null` and `undefined`).

There is an interesting difference between how the YouTube Iframe API handles setting the `opts` with an invalid `videoId` compared to setting the `loadVideoById` or `cueVideoById` with an invalid `videoId`; When setting the `videoId` through passing `opts` the YouTube API will load a "stopped" player regardless of the `autoplay` option whereas the `cueVideoById` will automatically start a video even if it is invalid resulting in the `an error occurred` message.

Long story short, I definitely think that we should make the `videoId` optional. As for loading the "stopped" player I'm not sure but this change at least creates some consistency.

:+1:

**Comment by goto-bus-stop** [comment]
Fixed style issues :v:

**Comment by tjallingt** [comment]
Great then now everything should be up-to-date again :smile:
Sorry for the delays by the way...
